### PR TITLE
Update super usage

### PR DESCRIPTION
**Pull request by rahulporuri**
On Python 3, `super` doesn't need to called with arguments where as on Python 2, `super` needs to be called with a class object and an instance as the two arguments e.g. `super(EArray, self)`.

Now that the pytables codebase is python 3 only, the `super` usage can be updated to remove the need to pass args.

Note that these changes were done using automated regex-based search and replace. `(super)+\(+(\w+)(,)+\s(self)+\)+`. The regex searches for the word `"super"`, followed by the character `(`, followed by another word, followed by the comma character `,`, followed by the word `"self"`, followed by the character `)`.

After the automated search and replace was done, each change was individually checked and added to the commit. Each change was checked to ensure that the `super` was being called with the current class and not a super class. For example, in `CArray`, the usage `super(Array, self)` has not been updated. A few other such uses have been observed. I am not sure if they are intentional or if they are hidden bugs.

Note to reviewer : These changes were not prompted by any specific issue in the repository but something I noticed while I was going through the codebase. This is a large changeset and I fully understand if the maintainer/maintainers decide to not merge this PR/close this PR/brake it up into smaller chunks.

Ref : https://docs.python.org/3/library/functions.html#super

### CODE REVIEW

**Comment by avalentino** [comment]
Hi @rahulporuri, thank you for the huge work.
Personally I'm all in favour of code modernization.
All tests seems to pass but I would like some other maintainer to commend on this before merging.

**Review by avalentino — APPROVED** [review]

**Comment by rahulporuri** [comment]
> Hi @rahulporuri, thank you for the huge work.
Personally I'm all in favour of code modernization.
All tests seems to pass but I would like some other maintainer to commend on this before merging.

I fully understand. Thanks for the review.

**Comment by FrancescAlted** [comment]
+1 I am onboard on this too!

**Review by FrancescAlted — APPROVED** [review]

**Comment by rahulporuri** [comment]
@FrancescAlted @avalentino I noticed these when I was making the changes and I want to bring them to your attention. I didn't include them in this PR because they are byclass the superclass. (I am not at all familiar with the internals of pytables so these might be intentional).

The automated search and replace made these three changes which I chose to not include because they are off. They are off because

* we call `super(Array, self)` inside the class `CArray`. Ref : https://github.com/PyTables/PyTables/blob/dd0e7e5f4d75cc26ee2e717f1580996f4fb9456e/tables/carray.py#L219-L220
* we call `super(Array, self)` inside the classes `CacheArray` and `IndexArray`. Ref : https://github.com/PyTables/PyTables/blob/dd0e7e5f4d75cc26ee2e717f1580996f4fb9456e/tables/indexesextension.pyx#L586 and https://github.com/PyTables/PyTables/blob/dd0e7e5f4d75cc26ee2e717f1580996f4fb9456e/tables/indexesextension.pyx#L1489

<details><summary>git diff of changes which werent included</summary>

```
diff --git a/tables/carray.py b/tables/carray.py
index 887cdc07..4e392c88 100644
--- a/tables/carray.py
+++ b/tables/carray.py
@@ -216,7 +216,7 @@ class CArray(Array):
                 self._v_chunkshape = tuple(SizeType(s) for s in chunkshape)

         # The `Array` class is not abstract enough! :(
-        super(Array, self).__init__(parentnode, name, new, filters,
+        super().__init__(parentnode, name, new, filters,
                                     byteorder, _log, track_times)

     def _g_create(self):
diff --git a/tables/indexesextension.pyx b/tables/indexesextension.pyx
index b076cd3d..f9d8a47c 100644
--- a/tables/indexesextension.pyx
+++ b/tables/indexesextension.pyx
@@ -583,7 +583,7 @@ cdef class CacheArray(Array):
     return

   def _g_close(self):
-    super(Array, self)._g_close()
+    super()._g_close()
     # Release specific resources of this class
     if self.mem_space_id > 0:
       H5Sclose(self.mem_space_id)
@@ -1486,7 +1486,7 @@ cdef class IndexArray(Array):


   def _g_close(self):
-    super(Array, self)._g_close()
+    super()._g_close()
     # Release specific resources of this class
     if self.mem_space_id > 0:
       H5Sclose(self.mem_space_id)
```

</details>

**Comment by avalentino** [comment]
@rahulporuri the call to `` super(Array, self).__init__(...)`` in ``carray.py`` is made on purpose (see also the comment on the previous line ``# The `Array` class is not abstract enough! :(``).

Regarding the two calls to ``super(Array, self)._g_close() `` in ``indexesextension.pyx``, IMHO that lines can be converted to ``super()._g_close()`` because ``Array`` don't seem to implement the ``_g_close`` method directly.
Also I tested that change an all seem to work.

In conclusion, please add a new commit with the updated ``super`` call for ``_g_close`` methods only (``indexesextension.pyx``).

**Comment by avalentino** [comment]
Thanks @rahulporuri
